### PR TITLE
Petting more animals now causes floating hearts and a moodlet, borgs can now pet animals.

### PR DESCRIPTION
**Pull request by uomo91**
## About The Pull Request

Renault, gondolas, penguins, cows, chicks, (small) lizards, rabbits, and sloths.
The borg hugging module now also works with this.

## Why It's Good For The Game

I think the extreme importance of this PR is self-evident.

This also cleans up some oversights from https://github.com/tgstation/tgstation/pull/52602

## Changelog
:cl:
tweak: Petting more animals than just cats and dogs now makes you happy.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm [hunk 1]
```diff
@@ -285,6 +304,25 @@
...
+			wuv(1,M)
+		if("harm")
+			wuv(-1,M)
+
+/mob/living/simple_animal/chick/proc/wuv(change, mob/M)
+	if(change)
+		if(change > 0)
+			if(M && stat != DEAD)
+				new /obj/effect/temp_visual/heart(loc)
+				emote("me", 1, "moos happily!")
```
Comment (EOBGames) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: dunno what kind of chicks you have where you're from, but I don't see them moo often

File: code/modules/mob/living/simple_animal/friendly/penguin.dm [hunk 2]
```diff
@@ -55,3 +55,22 @@
...
+/mob/living/simple_animal/pet/penguin/proc/wuv(change, mob/M)
+	if(change)
+		if(change > 0)
+			if(M && stat != DEAD)
+				new /obj/effect/temp_visual/heart(loc)
+				emote("me", 1, "honks happily!")
+				SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, src, /datum/mood_event/pet_animal, src)
+		else
+			if(M && stat != DEAD) // Same check here, even though emote checks it as well (poor form to check it only in the help case)
+				emote("me", 1, "crys and flaps its wings!")
```
Comment (Fikou) [inline code/modules/mob/living/simple_animal/friendly/penguin.dm:76]:
```suggestion
				emote("me", 1, "cries and flaps its wings!")
```

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm (hunk 1 above)
Comment (uomo91) [reply code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: Yeah, whoops.

File: code/modules/mob/living/simple_animal/friendly/penguin.dm (hunk 2 above)
Comment (uomo91) [reply code/modules/mob/living/simple_animal/friendly/penguin.dm:76]: Thanks.

File: code/modules/holiday/easter.dm [hunk 3]
```diff
@@ -60,6 +60,26 @@
...
+			wuv(1,M)
+		if("harm")
+			wuv(-1,M)
+
+/mob/living/simple_animal/chicken/rabbit/proc/wuv(change, mob/M)
+	if(change)
+		if(change > 0)
+			if(M && stat != DEAD)
+				new /obj/effect/temp_visual/heart(loc)
+				emote("me", 1, "hops around happily!")
```
Comment (MrDoomBringer) [inline code/modules/holiday/easter.dm:76]: Way too much repeated code here. You should really have this function defined on the /mob level, then add a bool var that controls whether or not this interaction happens or not, and set it to TRUE for these mobs. Add two string vars for the reaction they have (positive or negative) and you're all set.

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

File: code/modules/holiday/easter.dm (hunk 3 above)
Comment (uomo91) [reply code/modules/holiday/easter.dm:76]: I basically had this finished then ran into an enormous bug relating to mob emotes.
I'll try to keep working on this.

Comment (LemonInTheDark) [reply code/modules/holiday/easter.dm:76]: just keep working as if "me" works, I'll handle the rest

**Comment by LemonInTheDark** [comment]
Haha yes I have tricked uomo into working with me to improve the state of manual_emote(), I am truly evil....

**Comment by ExcessiveUseOfCobblestone** [comment]
> wuvable
> wuvemote

i dont care about local variables but ones on atoms/datums/proc args need to be as self-evident as possible so someone (see: admins/coders) who want to search for it have an easy time in doing so. Catspeak is not appropriate here.

File: code/game/objects/items/robot/robot_items.dm [hunk 4]
```diff
@@ -70,7 +70,9 @@
 	switch(mode)
 		if(0)
 			if(M.health >= 0)
-				if(user.zone_selected == BODY_ZONE_HEAD)
+				if(isanimal(M))
+					M.attack_hand(user)
+				else if(user.zone_selected == BODY_ZONE_HEAD)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/robot/robot_items.dm:75]: i think you want to early return and just have `if` instead of `else if`.

this proc lmao...

Comment (uomo91) [reply code/game/objects/items/robot/robot_items.dm:75]: Yeah I uh... wanted to spend as little time as humanly possible looking at that code.

I'll try that and see if it messes anything up.

Comment (uomo91) [reply code/game/objects/items/robot/robot_items.dm]: It seems to work.

**Pull request merged**